### ISSUE TITLE
Add support for grouping

### ISSUE THREAD

**Issue by deveth0** [Olen/homeassistant-plant#3]
It's quite useful to use a group with all plants to get the joined problems.
Sadly this does not work with this fork as the groups.py is missing:
https://github.com/home-assistant/core/blob/dev/homeassistant/components/plant/group.py

**Comment by Olen** [comment]
Hi,

I have not been able to follow up on keeping track of the original repo for this fork for a while.
We are still waiting for a policy change from Home Assistant to allow integrations to access (parts of) the configuration of other integrations so we can implement the Openplantbook API.

https://github.com/home-assistant/architecture/issues/459

**Comment by slaxor505** [comment]
It's fixed now by updating to the latest core-plant in this PR https://github.com/Olen/homeassistant-plant/pull/25

**Comment by Olen** [comment]
I have added groups.py to the new version, however, I do not think grouping of plants work, as I can't see an option to add a group of plants in the GUI.

**Comment by Olen** [comment]
I am pretty sure now, that we would need to add `plant` as a component here to be able to group them:
https://github.com/home-assistant/core/tree/dev/homeassistant/components/group 

I have tried to trick HA in several ways to think that it can group but no luck.

So I tried to basically copy the binary-sensor-related stuff in there to create plant-groups, and I am sort of getting there.  The group UI allows me to select the option "Plant", and add plant-entities, but I am still not able to actually create groups.

I'll close this for now.  "Someone" needs to first do all the stuff so HA can actually group entities under the "plant"-domain, then I can see what can be done.
